hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/BlockingRpcConnection.java

```diff
@@ -667,46 +685,34 @@ private void readResponse() {
         // this connection.
         int readSoFar = getTotalSizeWhenWrittenDelimited(responseHeader);
         int whatIsLeftToRead = totalSize - readSoFar;
+        LOG.debug("Unknown callId: " + id + ", skipping over this response of " + whatIsLeftToRead
+          + " bytes");
         IOUtils.skipFully(in, whatIsLeftToRead);
         if (call != null) {
           call.callStats.setResponseSizeBytes(totalSize);
-          call.callStats
-            .setCallTimeMs(EnvironmentEdgeManager.currentTime() - call.callStats.getStartTime());
         }
         return;
       }
-      if (responseHeader.hasException()) {
-        ExceptionResponse exceptionResponse = responseHeader.getException();
-        RemoteException re = createRemoteException(exceptionResponse);
-        call.setException(re);
-        call.callStats.setResponseSizeBytes(totalSize);
-        call.callStats
```

Comment:
it seems like you missed copying over the call to setCallTimeMs. Add that above the if branch?

Reply:
Just a refactoring follow the pattern in netty rpc connection. We will set the callTimeMs in AbstractRpcClient's call method, so we do not need to call setCallTimeMs here.

Reply:
Ok thanks, I see it there now